homework/zadanie4_five_philosophers.cpp

```diff
@@ -8,97 +8,102 @@
 
 using namespace std::chrono_literals;
 
-std::mutex mtx;
 
+class CFork{
+public:
+    mutable std::mutex mtx_fork_;
+};
+
+ 
 class CTable{
 
+    int numbers_of_forks_;
+
 public:
-    std::vector<bool> forks_pool;
+    std::vector<CFork*> forks_pool_;
```

Comment:
Wektor wskaźników mnie trochę martwi. Ktoś inny zarządza tą pamięcią i trzeba pamiętać o jej zaalokowaniu i zwolnieniu. `std::vector<CFork>` wydaje mi się bardziej naturalnym i mniej błędogennym rozwiązaniem ;) Wtedy konstruktor nie musi nic alokować, a dealokować. Tylko wywołasz konstruktor wektora na liście inicjalizacyjnej przekazując mu liczbę sztućców.
I dlaczego to w sekcji public? :)

Reply:
Vector forkow jest teraz w stole. i nie sa w sekcji public:)

Reply:
ale usuwanie dorzucilem do destruktora.

Reply:
jesli nie byloby wskaznikow to jakbym mial z palca dorucic 100 forkow? troche bez sensu jak dla mnie, a tak mozemy sobie to w szybki sposob zmienic.

Reply:
vector forkow juz nie jest zbiorem wskaznikow,